### PR TITLE
[Layer] Refactoring Layers class

### PR DESCRIPTION
**Pull request by again4you**
* Separate single Layers file into several files for its role.
* Cleanup unused headers and internal functions
* Rename files name (layers.[h|cpp] -> layer.[h|cpp])
* Increase patch verson of debian package since header files are added

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/include/layer.h [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  /**
+   * @brief     Forward Propation of neural Network
+   * @param[in] in Input Tensor taken by upper layer
+   * @retval    Output Tensor
+   */
+  virtual Tensor forwarding(Tensor in, int &status) = 0;
+
+  /**
+   * @brief     Forward Propation of neural Network
+   * @param[in] in Input Tensor taken by upper layer
```
Comment (kparichay) [inline nntrainer/include/layer.h:105]: add `output` entry as well?

File: nntrainer/include/bn_layer.h [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+   * @retval    Activation(W x input + B)
+   */
+  Tensor forwarding(Tensor in, int &status);
+
+  /**
+   * @brief     foward propagation : return Input Tensor
+   *            It return Input as it is.
+   * @param[in] input input Tensor from lower layer.
+   * @param[in] output label Tensor.
+   * @retval    Activation(W x input + B)
```
Comment (kparichay) [inline nntrainer/include/bn_layer.h:76]: retval description is strange for batch normalization layer.

File: nntrainer/include/bn_layer.h [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+   *            5. epsilon : float
+   */
+  enum class PropertyType {
+    input_shape = 0,
+    bias_zero = 1,
+    epsilon = 5,
+  };
+
+private:
+  Tensor weight;
```
Comment (kparichay) [inline nntrainer/include/bn_layer.h:140]: I think bn layer dont need weight or bias.

Comment (jijoongmoon) [reply nntrainer/include/bn_layer.h:140]: yes. bn layer needs verification deeply with test cases and application. But not this PR I think.

Comment (kparichay) [reply nntrainer/include/bn_layer.h:140]: ok.

**Comment by again4you** [comment]
### Submit v2
* Update the comment for forwarding() method of BatchNormalizationLayer class
  `* @retval    normalized input tensor using scaling factor`
* Add omitted param comment for forwarding() method of Layer class

File: Applications/Classification/jni/main_func.cpp [hunk 4]
```diff
@@ -33,7 +33,7 @@
 #include <stdlib.h>
 #include <time.h>
 
-#include "layers.h"
+#include "layer.h"
```
Comment (jijoongmoon) [inline Applications/Classification/jni/main_func.cpp:36]: I think you can delete it.

File: Applications/Classification/jni/main.cpp [hunk 5]
```diff
@@ -39,7 +39,7 @@
 #include <stdlib.h>
 #include <time.h>
 
-#include "layers.h"
+#include "layer.h"
 #include "neuralnet.h"
```
Comment (jijoongmoon) [inline Applications/Classification/jni/main.cpp:43]: I think you can delete it.
